df_chat/tests/test_message_endpoint.py

```diff
@@ -26,6 +31,110 @@ def test_message_creation_success(self):
         self.assertEqual(message["room_user_id"], room_user.pk)
         self.assertIsNone(message["parent_id"])
         self.assertFalse(message["is_reaction"])
+        self.assertEqual(response.status_code, http.HTTPStatus.CREATED)
+
+        return user, room
+
+    async def test_filter_message(self):
+        """
+        Test-case for filtering messages.
+        The purpose of this is test is to perform multiple filters on the
+            message list API. These filters include:
+        1. Filter by `body` field (keyword sear ch).
+        2. Filter by `User`'s `username` field.
+        3. Filter by `created` field (datetime range).
+        In-order to do so, we test multiple cases against the message created
+            in the `test_create_message` test.
+        """
+        user, room = await self.test_message_creation_success()
+
+        messages_list_api_url = f"/api/v1/chat/rooms/{room.id}/messages/"
+
+        # * 1. Keyword search - ok cases
+        for body in [
+            "Test message 1234.",
+            "Test message 1234",
+            "Test",
+            "test",
+            "Message",
+            "message",
+            "1234",
+        ]:
+            messages_list_response = await self.async_client.get(
+                messages_list_api_url,
+                {"body": body},
+            )
+            # Assert the correct HTTP status code for response.
+            self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+            # Assert the response body contains the keyword.
+            self.assertIn(
+                body.lower(), messages_list_response.json()[0]["body"].lower()
+            )
+
+        # * 1. Keyword search - not ok cases
+        for body in [
+            "somethingrandom",
+            "messages",
+            "124",
+        ]:
+            messages_list_response = await self.async_client.get(
+                messages_list_api_url,
+                {"body": body},
+            )
+            # Assert the correct HTTP status code for response.
+            self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
```

Comment:
Please asset that the messages_list_response.json() is empty list, as these keywords are not present in any of the messages.